pkg/storaged/dialogx.jsx

```diff
@@ -462,3 +464,128 @@ export const Skip = (className, options) => {
         }
     };
 };
+
+const StatelessSlider = ({ fraction, onChange }) => {
+    function start_dragging(event) {
+        let el = event.currentTarget;
+        let width = el.offsetWidth;
+        let left = el.offsetLeft;
+        while (el.offsetParent) {
+            el = el.offsetParent;
+            left += el.offsetLeft;
+        }
+
+        function drag(event) {
+            let f = (event.pageX - left) / width;
+            if (f < 0) f = 0;
+            if (f > 1) f = 1;
+            onChange(f);
+        }
+
+        function stop_dragging() {
+            document.removeEventListener("mousemove", drag);
+            document.removeEventListener("mouseup", stop_dragging);
+        }
+
+        document.addEventListener("mousemove", drag);
+        document.addEventListener("mouseup", stop_dragging);
+        drag(event);
+    }
+
+    if (fraction < 0) fraction = 0;
+    if (fraction > 1) fraction = 1;
+
+    return (
+        <div className="slider" onMouseDown={start_dragging}>
+            <div className="slider-bar" style={{ width: fraction * 100 + "%" }}>
+                <div className="slider-thumb" />
+            </div>
+        </div>
+    );
+};
+
+class SizeSliderElement extends React.Component {
+    constructor(props) {
+        super();
+        this.units = cockpit.get_byte_units(props.value || props.max);
+        this.units.forEach(u => { if (u.selected) this.state = { unit: u.factor }; });
+    }
+
+    render() {
+        let { val, max, onChange } = this.props;
+        let { unit } = this.state;
+
+        const change_slider = (f) => {
+            onChange(Math.round(f * max));
+        };
+
+        const change_text = (event) => {
+            if (event.type == "change") {
+                let val = +event.target.value * unit;
```

Comment:
This looks a bit odd -- did the + just slip in through fat-fingering, or did you really mean to add something to e.t.value?

Reply:
This is a JS idiom for converting an arbitrary value into a number.  `+x` is equivalent to `Number(x)`, so maybe I should use the latter.

Values are implicitly converted to numbers as soon as you compute with them, so it isn't trictly necessary to do it explicitly, but I think it is clearer.  Except when it isn't. :)

Reply:
I see, so this takes care of translating `''` (empty string) and `null` to `0`. (It doesn't work for `undefined` though). Thanks, I'll remember this.